torch_audiomentations/augmentations/background_noise.py

```diff
@@ -0,0 +1,53 @@
+import random
+import torch
+from ..core.transforms_interface import BasicTransform
+from ..utils.file import find_audio_files, load_audio
+from ..utils.dsp import calculate_rms, calculate_desired_noise_rms
+
+
+class ApplyBackgroundNoise(BasicTransform):
+    """
+    Applies a background noise to the input signal.
+    """
+
+    def __init__(self, bg_path, snr_in_db, p=0.5):
+        super(ApplyBackgroundNoise, self).__init__(p)
+        self.bg_path = find_audio_files(bg_path)
+        assert len(self.bg_path) > 0
+        self.snr_in_db = snr_in_db
+
+    def randomize_parameters(self, samples, sample_rate):
+        super(ApplyBackgroundNoise, self).randomize_parameters(samples, sample_rate)
+        if self.parameters["should_apply"] and samples.size(0) > 0:
+            bg_file_paths = random.choices(self.bg_path, k=samples.size(0))
+            bg_audios = []
+            for index, bg_file_path in enumerate(bg_file_paths):
+                bg_audio = torch.from_numpy(load_audio(bg_file_path, sample_rate))
+                max_bg_offset = max(0, bg_audio.size(0) - samples[index].size(0) - 1)
+
+                bg_start_index = random.randint(0, max_bg_offset)
+                bg_end_index = bg_start_index + samples[index].size(0)
+                bg_audios.append(bg_audio[bg_start_index:bg_end_index])
```

Comment:
In the case where noise files are much longer that the input, this doesn't look very efficient. 
`soundfile` supports accessing the length of the audio without loading and loading only a portion of it, which would be more efficient.

Reply:
Resolving at this moment since latest commits take advantage of `soundfile`'s audio chunk loading to optimize the performance. Conditions are checked whether to repeat and truncate or not so that the appropriate steps are performed.